### PR TITLE
Added Pants.gitignore for Pants

### PR DESCRIPTION
**Pull request by N0K0**
# Pull Request

Thank you for contributing to @toptal/gitignore and https://www.gitignore.io.

## New or update

Select the appropriate check box for this pull request. This helps when merging to ensure there are no conflicts with other templates or misunderstandings of how thee template list works.

### New

- [x] Template - New `.gitignore` template
- [ ] Composition - Template made from smaller templates
- [ ] Inheritance - Template similar to an existing template
- [ ] Patch - Template extending functionality of existing template

### Update

- [ ] Template - Update existing `.gitignore` template

## Details

Pants is a monorepo build system like for example Gradle or Bazel
Added the workspace files as per their docs
https://www.pantsbuild.org/docs/initial-configuration#update-gitignore

### CODE REVIEW

File: templates/Pants.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Pants workspace files
+/.pants.d/
+/dist/
+/.pids
+/.pants.workdir.file_lock*
```
Comment (konstrybakov) [inline templates/Pants.gitignore:5]: Hey, @N0K0, thank you for the contribution! In the documentation, I see this line not being added to `.gitignore` with the latest version of the Pants (@2.14). Do you believe it's still useful to add it? Why?

Comment (N0K0) [reply templates/Pants.gitignore:5]: Interesting! Looks like is not needed after 2.10. Will edit during the weekend :)

**Comment by N0K0** [comment]
@konstrybakov, updated to match the 2.14 format, any thoughts on how we could potentially support the older ones too? :)

**Review by konstrybakov — APPROVED** [review]

**Comment by konstrybakov** [comment]
@N0K0 There isn't a mechanism in place to have versioning for gitignore files ATM! But if you can come up with something, feel free to open up a PR :)

**Pull request merged**